Clockwise/Budget.cs

```diff
@@ -24,27 +24,25 @@ public abstract class Budget
             StartTime = Clock.Now();
         }
 
-        internal IClock Clock { get; }
-
-        public TimeSpan ElapsedDuration => Clock.Now() - StartTime;
-
-        public bool IsExceeded =>
-            CancellationTokenSource.IsCancellationRequested;
-
         public CancellationToken CancellationToken => CancellationTokenSource.Token;
 
-        public IReadOnlyCollection<BudgetEntry> Entries => entries.OrderBy(e => e.ElapsedDuration).ToArray();
+        protected CancellationTokenSource CancellationTokenSource { get; }
 
-        public DateTimeOffset StartTime { get; }
+        internal IClock Clock { get; }
```

Comment:
Having this on the base class is a little weird but it's internal so no one will take a dependency on it. This will probably get cleaned up later.